esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -3788,14 +3787,17 @@ void CWsWuFileHelper::createZAPWUGraphProgressFile(const char* wuid, const char*
     writeToFile(fileName.str(), graphProgressXML.length(), graphProgressXML.str());
 }
 
-int CWsWuFileHelper::zipAFolder(const char* folder, const char* passwordReq, const char* zipFileNameWithPath)
+int CWsWuFileHelper::zipZAPFolders(const char* zapWorkingFolder, StringArray& localFiles, const char* passwordReq, const char* zipFileNameWithPath)
```

Comment:
>int CWsWuFileHelper::zipZAPFolders(const char* zapWorkingFolder, StringArray& localFiles, const char* passwordReq, const char* zipFileNameWithPath)


it would be better if the method return the zip file it created, i.e. a prototype like:
```
int CWsWuFileHelper::zipZAPFolders(const char* tempFolder, StringArray& localFiles, const char* passwordReq, StringBuffer &zipFilePath)
```

Afaics, the ultimate user of the zip file (callers to createWUZAPFile), don't create about the name of the temporary zip file, which will be read as a byte stream, e.g. in the ::onWUCreateZAPInfo case, the ZAPFileName response can be set to something more friendly, it doesn't have to be the same as the temp zip filename zipZAPFolders created.

Reply:
If a zapFileName is set in user request, the existing setZAPFile() does set the zip file name based on the request. If not set, I may change the zap report to wuid.zip.

Reply:
the user can request whatever filename they want chrome to save, that is set into the response ("attachment;filename=...."), but the filename we create/save on the server doesn't have to be the same.
i.e. you can choose a name base on the unique temp folder created (<unique-folder>.zip), but then set attachment;filename=<user-supplied-name>